tests/common/config_reload.py

```diff
@@ -56,19 +56,20 @@ def config_force_option_supported(duthost):
 
 
 @ignore_loganalyzer
-def config_reload(duthost, config_source='config_db', wait=120, start_bgp=True, start_dynamic_buffer=True,
+def config_reload(sonic_host, config_source='config_db', wait=120, start_bgp=True, start_dynamic_buffer=True,
                   safe_reload=False, wait_before_force_reload=0,
-                  check_intf_up_ports=False, traffic_shift_away=False, override_config=False):
+                  check_intf_up_ports=False, traffic_shift_away=False, override_config=False, is_dut = True):
     """
     reload SONiC configuration
-    :param duthost: DUT host object
+    :param sonic_host: SONiC host object
     :param config_source: configuration source is 'config_db', 'minigraph' or 'running_golden_config'
-    :param wait: wait timeout for DUT to initialize after configuration reload
+    :param wait: wait timeout for sonic_host to initialize after configuration reload
     :param override_config: override current config with '/etc/sonic/golden_config_db.json'
+    :param is_dut: True if the host is DUT, False if the host may be neighbor device
```

Comment:
Can you elaborate more about why we need is_dut as a special parameter?
Like: If the host is a neighbor device, it may lack of some runtime configurations like mg_facts....